code/modules/mob/living/simple_animal/bot/ed209bot.dm

```diff
@@ -199,12 +214,13 @@ Auto Patrol[]"},
 	if(disabled)
 		return
 
+	var/judgement_criteria = judgement_criteria()
 	var/list/targets = list()
 	for(var/mob/living/carbon/C in view(7,src)) //Let's find us a target
 		var/threatlevel = 0
 		if((C.stat) || (C.lying))
 			continue
-		threatlevel = C.assess_threat(src, lasercolor)
+		threatlevel = C.assess_threat(judgement_criteria, lasercolor, weaponcheck=CALLBACK(src, .check_for_weapons))

```

Comment:
.procname only works on new procs from inside new procs, other wise it works on overrides (from inside overrides (and i think new procs)

Welcome to byond.